### PR TITLE
Linkerd 2.10

### PR DESCRIPTION
**Pull request by ubergesundheit**
TODO:

- [x] Changelog entry

### CODE REVIEW

File: helm/linkerd2-cni-app/Chart.yaml [hunk 1]
```diff
@@ -20,4 +20,6 @@ name: linkerd2-cni-app
 sources:
 - https://github.com/linkerd/linkerd2
 - https://github.com/giantswarm/linkerd2-cni-app
-version: [[ .Version ]]
+version: 0.3.1-beta-fc0798077fdec63274b019706e9c08c861b82d98
```
Comment (webwurst) [inline helm/linkerd2-cni-app/Chart.yaml:23]: Should it be that version? In general releasing a `beta` would be fine for me.

**Comment by webwurst** [comment]
Lots of changes. Do we need to provide an upgrade path for the customers? Is our linkerd-app already used in production somewhere?

File: helm/linkerd2-cni-app/Chart.yaml (hunk 1 above)
Comment (ubergesundheit) [reply helm/linkerd2-cni-app/Chart.yaml:23]: oh no, good catch that is a leftover from failed `abs` runs

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 workflows:
   package-and-push-chart-on-tag:
     jobs:
       - architect/push-to-app-catalog:
           context: "architect"
+          executor: "app-build-suite"
           name: "package and push linkerd2-cni-app chart"
-          app_catalog: "giantswarm-playground-catalog"
-          app_catalog_test: "giantswarm-playground-test-catalog"
+          app_catalog: "giantswarm-catalog"
```
Comment (ubergesundheit) [inline .circleci/config.yml:12]: Towards https://github.com/giantswarm/giantswarm/issues/15706

**Comment by ubergesundheit** [comment]
> Do we need to provide an upgrade path for the customers? Is our linkerd-app already used in production somewhere?

I think this is not the case, so no upgrade path required.

**Review by webwurst — APPROVED** [review]

**Pull request merged**